Algorithms/Dynamic-Programming/Longest Increasing Subsequence.cpp

```diff
@@ -4,56 +4,56 @@
 //                    in increasing order.
 // Input Format : The first line contains n,the number of elements of the array.
 //                The second line contains n space separated integers.
-// Output Format : A single number specifying the length of LIS, followed by each element of LIS.
+// Output Format : A single number specifying the length of LIS.
 
 // Approach Used : Dynamic Programming
 
 // Intuition behind dynamic programming:
-// In this solution each element of the aux vector ,say at index i stores the LIS possible with the 
-// last element arr[i](ith element of the input array). So clearly, the first element of aux stores just arr[0].
+// In this solution each element of the aux vector ,say at index i stores the length of LIS possible with the 
+// last element arr[i](ith element of the input array). So clearly, the first element of aux stores just 1(only arr[0]).
 // Next for finding LIS for other elements as last element of that sequence, an iteration is made for
 // all the previous LIS whose last element is less than the current element. If such a sequence is found
-// the largest of these is taken with the last element appended at the end of it. Finally the sixe of the maximum sequence
-// along with the sequence is printed. The memorisation of previous LIS is a key to dynamic programming here and forms the
-// overlapping subproblems.
+// the largest of these is taken with the last element appended at the end of it(one is added actually to realise it). Finally the 
+// size of the maximum sequence is printed. The memoization of previous LIS length is a key to dynamic programming here 
+// and is the technique to store the solutions of subproblems.
  
 #include<iostream>
 #include<vector>
 using namespace std;
 
 
-void calLIS(vector<int>arr)
+void calLIS(vector<int>arr,int n)
```

Comment:
`n` need not be taken as input to the function since `arr.size()` will provide with this value.

Reply:
Done!